receiver/statsdreceiver/receiver.go

```diff
@@ -111,12 +111,16 @@ func (r *statsdReceiver) Start(ctx context.Context, _ component.Host) error {
 				for _, batch := range batchMetrics {
 					batchCtx := client.NewContext(ctx, batch.Info)
 
-					if err := r.Flush(batchCtx, batch.Metrics, r.nextConsumer); err != nil {
+					err := r.Flush(batchCtx, batch.Metrics, r.nextConsumer)
+					r.reporter.RecordFlushedMetrics(int64(batch.Metrics.MetricCount()), err)
```

Comment:
Looking at the places with metrics instrumentation, this should be number of datapoints not the number of metrics ref. https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e9575417614791c3362b00fcafd56393217ef742/receiver/prometheusreceiver/internal/transaction.go#L305

Reply:
Good point.  Do you want to make the suggested changes?  I see you have a related PR open https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31822